src/contracts/GPv2Settlement.sol

```diff
@@ -342,21 +348,32 @@ contract GPv2Settlement is GPv2Signing, ReentrancyGuard, StorageAccessible {
         }
     }
 
-    /// @dev Claims order gas refunds by freeing storage for all encoded order
-    /// gas refunds.
+    /// @dev Claims order gas refunds.
+    ///
+    /// @param orderRefunds Order refund data for freeing storage.
+    function claimOrderRefunds(OrderRefunds calldata orderRefunds) internal {
+        freeOrderStorage(orderRefunds.filledAmounts, filledAmount);
+        freeOrderStorage(orderRefunds.preSignatures, preSignature);
+    }
+
+    /// @dev Claims refund for the specified storage and order UIDs.
     ///
     /// This method reverts if any of the orders are still valid.
     ///
-    /// @param orderRefunds Unique identifiers of orders to get a gas refund for
-    /// by freeing storage.
-    function claimOrderRefunds(bytes[] calldata orderRefunds) internal {
-        for (uint256 i = 0; i < orderRefunds.length; i++) {
-            bytes calldata orderUid = orderRefunds[i];
+    /// @param orderUids Order refund data for freeing storage.
+    /// @param orderStorage Order storage mapped on a UID.
+    function freeOrderStorage(
+        bytes[] calldata orderUids,
+        mapping(bytes => uint256) storage orderStorage
```

Comment:
Nice! Does this line have any impact on gas cost? What's actually going on in the EVM? Maybe the storage slot that identifies the mapping is the actual "input" of the jump?

Reply:
It looks like its passing the storage slot index (which is used for `keccak(slotIndex || orderUid)` to compute the actual slot).